enhancements/ingress/set-delete-http-headers.md

```diff
@@ -0,0 +1,1031 @@
+  Answer: A common technique is to put static assets on a different virtual host. In OpenShift terms, that means using a distinct route for static assets. This route can
+  be annotated with `haproxy.router.openshift.io/disable_cookies=true` to prevent setting the cookie and thus prevent setting cache-control.
+
+#### Would we be able to [Replace a header by using a regular expression](https://www.haproxy.com/documentation/hapee/latest/traffic-routing/rewrites/rewrite-responses/#replace-a-header-by-using-a-regular-expression) or [Replace part of a header by using a regular expression](https://www.haproxy.com/documentation/hapee/latest/traffic-routing/rewrites/rewrite-responses/#replace-part-of-a-header-by-using-a-regular-expression)?
+  Answer: No, this is out of scope for this enhancement. Action `Set` will perform `set-header` in the haproxy.config for `http-response` and `http-request` which will either set the value if the value is not present or replace the existing value.
+
+#### Should this enhancement allow setting dynamic header values?
+  Answer: Yes, you would be able to set a minimal set of dynamic header values. 
+  We had three options for this EP.
+  1. Allow dynamic header values with all sample fetchers and converters.
+  2. Allow a minimal set of dynamic values that we can confidently validate, focusing on the things that people have asked for on the RFE or [EP](https://github.com/openshift/enhancements/pull/1296#issuecomment-1396103783).
+  3. Prohibit dynamic values entirely.
+
+  However, as few customers asked for it, we went ahead with option number 2.
+
+  For setting dynamic header values for HTTP responses, allowed sample fetchers are `res.hdr` and `ssl_c_der`. Allowed converters are `lower` and `base64`.
+  
+  For setting dynamic header values for HTTP requests, allowed sample fetchers are `req.hdr` and `ssl_c_der`. Allowed converters are `lower` and `base64`.
+    
+  Example: One might be able to set the value of `x-forwarded-client-cert` to `%{+Q}[ssl_c_der,base64]`.
+
+#### Certain headers like HSTS are not allowed to be set using this API as there is already a route annotation along with a setting that an admin can add to a domain via spec.requiredHSTSPolicies
+#### as explained in [Enforcing hsts per domain route configuration](https://docs.openshift.com/container-platform/4.13/networking/routes/route-configuration.html#nw-enforcing-hsts-per-domain_route-configuration) available for this
+#### Would it not be better to provide a concrete API and deprecate the annotation API?
+  Answer: The strict-transport-security header (AKA HSTS) has wider ranging implications than the typical HTTP header. We considered allowing HSTS to be configured
+  on the IngressController in the past; see the global-options-enable-hsts EP, which was never implemented because it was superseded by
+  https://github.com/openshift/enhancements/blob/master/enhancements/ingress/global-admission-hsts.md, which was implemented.
```

Comment:
```suggestion
  Answer: The `strict-transport-security` header (for HSTS) has wider ranging implications than the typical HTTP header. We considered allowing HSTS to be configured
  on the IngressController in the past; see the global-options-enable-hsts EP, which was not accepted, and it's supercedent
  https://github.com/openshift/enhancements/blob/master/enhancements/ingress/global-admission-hsts.md, which was implemented.
```

Reply:
done

Reply:
It seems to me that the answer is yes, we should, but doing so is beyond the scope of this EP.  Replacing the annotation with a formal API was discussed here: https://github.com/openshift/enhancements/pull/1296#discussion_r1082528269, and Joel and David both seem to favor doing so, but as Miheer said, replacing the annotation is a significant amount of work and goes beyond the requirements for this EP.